test/framework/src/main/java/org/elasticsearch/test/disruption/DisruptableMockTransport.java

```diff
@@ -254,8 +256,10 @@ public String toString() {
         }
     }
 
-    private NamedWriteableRegistry writeableRegistry() {
-        return new NamedWriteableRegistry(ClusterModule.getNamedWriteables());
+    public static NamedWriteableRegistry writeableRegistry() {
```

Comment:
We need the named writables from `NetworkModule` for the force empty primary allocation calls added to the tests.

Reply:
Instead of statically collecting them here, let's make this a protected method, lift it to MockTransport and override it in SnapshotsServiceTests

Reply:
Done in https://github.com/elastic/elasticsearch/pull/37686/commits/803003f92f3d2a448719c2dbd04cd51256e9bb44 :)